engines/docker/sandboxbuilder.go

```diff
@@ -0,0 +1,120 @@
+package dockerengine
+
+import (
+	"context"
+	"regexp"
+	"sync"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+)
+
+type sandboxBuilder struct {
+	engines.SandboxBuilderBase
+	command    []string
+	image      imageType
+	imageDone  chan struct{}
+	imageError error
+	monitor    runtime.Monitor
+	e          *engine
+	env        *docker.Env
+	taskCtx    *runtime.TaskContext
+	discarded  bool
+	cancelPull context.CancelFunc
+	mu         sync.Mutex
+	handle     *caching.Handle
+}
+
+func newSandboxBuilder(payload *payloadType, e *engine, monitor runtime.Monitor,
+	ctx *runtime.TaskContext) *sandboxBuilder {
+	sb := &sandboxBuilder{
+		command:   payload.Command,
+		image:     payload.Image,
+		monitor:   monitor,
+		e:         e,
+		taskCtx:   ctx,
+		env:       &docker.Env{},
+		imageDone: make(chan struct{}, 1),
+	}
+	// set image
+	sb.image.engine = sb.e
+	pctx, cancelPull := context.WithCancel(context.Background())
```

Comment:
We don't need this... Use the `ctx` parameter... it'll be cancelled before `Discard()` is called.